osbenchmark/utils/parse.py

```diff
@@ -0,0 +1,59 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+
+def parse_string_parameter(key: str, params: dict, default: str = None) -> str:
+    if key not in params:
+        if default is not None:
+            return default
+        raise ConfigurationError(
+            "Value cannot be None for param {}".format(key)
+        )
+
+    if isinstance(params[key], str):
+        return params[key]
+
+    raise ConfigurationError("Value must be a string for param {}".format(key))
+
+
+def parse_int_parameter(key: str, params: dict, default: int = None) -> int:
```

Comment:
What happens if we pass integer value to string parameter?  The above looks complicated for simple function. I don't see any advantage. However, if you have strong opinion, i can refactor.

Reply:
> What happens if we pass integer value to string parameter? 
Are you referring to passing an integer value to the string parameter `key`? The code above is based off of the three functions (`parse_str_parameter`, `parse_float_parameter`, and `parse_int_parameter`) and how they were used in the unittests in `parse_test.py`. In `parse_test.py`, it doesn't seem like there's a scenario where an integer value is passed to `key` since only values such as "int-value", "string-value", "default-value", "float-value" are provided. I was under the assumption that OSB must first identify if the parameter is one of the three valid datatypes and then call the function matching that data type. Let me know if I'm misunderstanding this. 

That being said, I don't have a strong preference and we can keep these functions as is 👍🏻